src/main/java/com/denizenscript/depenizen/bukkit/events/breweryx/BreweryChatDistortScriptEvent.java

```diff
@@ -0,0 +1,84 @@
+package com.denizenscript.depenizen.bukkit.events.breweryx;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.PlayerTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.depenizen.bukkit.objects.breweryx.BreweryPlayerTag;
+import com.dre.brewery.api.events.PlayerChatDistortEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class BreweryChatDistortScriptEvent extends BukkitScriptEvent implements Listener {
+
+
+    // <--[event]
+    // @Events
+    // brewery chat distort
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a drunk brewery player's chat is distorted.
+    //
+    // @Context
+    // <context.message> Returns the message to be sent after being distorted.
+    // <context.original_message> Returns the original message before being distorted.
+    // <context.bplayer> Returns an BPlayerTag of the player that drank the brew.
+    // <context.player> Returns a PlayerTag of the player that had their chat distorted.
+    //
+    // @Determine
+    // ElementTag(String) to set the message to be sent after being distorted.
+    //
+    // @Plugin Depenizen, BreweryX
+    //
+    // @Player Always.
+    //
+    // @Group Depenizen
+    //
+    // -->
+    public BreweryChatDistortScriptEvent() {
+        registerCouldMatcher("brewery chat distort");
+    }
+
+    public PlayerChatDistortEvent event;
+    public PlayerTag playerTag;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        return super.matches(path);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(event.getPlayer());
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "message" -> new ElementTag(event.getDistortedMessage());
+            case "original_message" -> new ElementTag(event.getWrittenMessage());
+            case "bplayer" -> BreweryPlayerTag.forPlayer(playerTag);
+            case "player" -> playerTag;
+            default -> super.getContext(name);
+        };
+    }
+
+    @Override
+    public boolean handleDetermination(ScriptPath path, String prefix, ObjectTag value) {
+        if (prefix.equals("message") && value instanceof ElementTag elementTag) { // FIXME - Cast redundant but .toString() returns class and hash?
```

Comment:
... what?

Reply:
I think I might just be confused here. If I don't cast the ObjectTag to an ElementTag I can't do value.asString(). When you told me the cast was redundant I assumed you meant I could just do ObjectTag#toString(). Just looked again and I'm guessing you wanted me to do `value.asElement().asString()`?

Reply:
I'm confused why you think toString doesn't work but yes asElement().asString() is the most proper option

Reply:
Going to the declaration of value.toString() brought me to the Object.java class which is why I why I figured it wouldn't work. Sorry about that